### DISCUSSION TITLE
Msre example problem

### DISCUSSION THREAD

**Discussion by saqibhayat556** [arfc/moltres#244] [Q&A Reactor multiphysics] [answered]
Hello @smpark7 Sir! 
Msre input file, i-e  auto_diff_rho.i, works fine with the given cross-sections but when i use cross-sections generated with OpenMc. I'm not getting the right results. Is there any problem with my cross-sections or input file? guide me please. Files are attached below. One more question do we need to give burnup information when generating cross-sections?
[input.txt](https://github.com/arfc/moltres/files/12448392/input.txt)
[Xs.txt](https://github.com/arfc/moltres/files/12448401/Xs.txt)

![Screenshot from 2023-08-27 22-17-46](https://github.com/arfc/moltres/assets/124503670/b1b124c4-807b-4844-ba74-73d59afe8838)

**Comment by smpark7** [comment]
Your results show that the flux is decreasing to zero. That probably means that with your cross sections, the reactor is subcritical at 922K. What is the k_eff value from the OpenMC calculations? You'll need to figure out the approximate isothermal temperature at which the system is critical and ensure that your inlet temperature is lower than that temperature value. You should probably also expand the temperature range of your cross sections to ensure that your simulation stays within the given range.

**Reply by saqibhayat556** [reply]
Thanks @smpark7! I have modeled 2D unit cell of MSRE with reflective boundary condition to generate cross sections for 3 temperatures, 922, 972, and 1022, as used for msre in property_file_dir. At 922K keff comes out to be around 1.56 for this unit cell. I don't know whether whole core would be critical or not. I have taken dimensions of this unit cell from thesis ''Multi physics simulations of Molten Salt
Reactors using the Moltres code''. You are probably right because when i use these cross sections with the input file given below with different boundary condition for temperature at inlet and outer wall. My simulation works fine but the issue is when i take number of time steps equals to 100 my multiplication factor at final step approaches 1.000062 but when i increase number of time steps more then 105  my simulation does not converge. Error is reported below. I was wondering what the issue could be?
![Screenshot from 2023-08-29 01-30-02](https://github.com/arfc/moltres/assets/124503670/1eeee743-6e85-4052-aceb-a72c85acba73)
[input.txt](https://github.com/arfc/moltres/files/12458025/input.txt)

**Reply by smpark7** [reply]
Can you explain to me what you're trying to do in this input file? I see that you made some changes but it's difficult to read through your file without the indents.

**Comment by saqibhayat556** [comment]
sorry for the inconvenience! I am trying to run MSRE example with cross sections generated in OpenMc. The problem is, if i increase number of time steps more then 105 my simulation fails. Why does the simulation fail to converge? Is it because of errors in the cross sections? Below cross section file is also attached with input file. Up to 104 number of time steps simulation works fine. 
[input.txt](https://github.com/arfc/moltres/files/12458516/input.txt)
[Xs.txt](https://github.com/arfc/moltres/files/12458519/Xs.txt)

**Reply by smpark7** [reply]
When I ran your input file, the flux values decreased over time until the flux values become too small for the residual values to meet the convergence criteria. Your problem is still sub-critical. Assuming your cross-sections are representative of a molten salt reactor, you need to raise the initial and inlet temperature or increase the uranium content in your OpenMC simulation.

**Reply by saqibhayat556** [reply]
Thanks @smpark7! I will look into it. How much excess reactivity should I add to my geometry by increasing the uranium content when simulating in OpenMC, such that my problem would become critical when modeling in Moltres at the same temperature? For instance, if my geometry was exactly critical at 9% enrichment in OpenMC, how much should I raise its enrichment? Should I increase it to 10%?

**Reply by smpark7** [reply, marked as answer]
You'll have to investigate this on your own by trial and error, especially since you're using a unit cell representation to generate the cross sections. For k_eff calculations with Moltres, you can refer to the `tests/nts/nts_no_action.i` or `problems/2021-cnrs-benchmark/phase-0/nts.i` input files to run your model and get an idea of what the k_eff is for your current model.

**Reply by saqibhayat556** [reply]
Thank you!
